packages/next/build/webpack/plugins/flight-types-plugin.ts

```diff
@@ -25,8 +25,9 @@ type TEntry = typeof entry
 
 check<IEntry, TEntry>(entry)
 
-type PageProps = { params?: any }
-type LayoutProps = { children: React.ReactNode; params?: any }
+type PageParams = Record<string, string>
+type PageProps = { params?: PageParams, searchParams?: Record<string, string | string[]> }
+type LayoutProps = { children: React.ReactNode; params?: PageParams }
```

Comment:
```suggestion
interface PageProps {
  params?: PageParams
  searchParams?: Record<string, string | string[]>
}
interface LayoutProps {
  children: React.ReactNode
  params?: PageParams
}
```

I would prefer interface over type, [as it has better performance](https://github.com/microsoft/TypeScript/wiki/Performance#preferring-interfaces-over-intersections).